src/usr/share/batwatch/batwatch.py

```diff
@@ -53,124 +46,150 @@ def __eq__(self, other):
+                else:
+                    self.logger.info('Battery state changed from %s to %s.', prior_status,
+                                     current_status)
+
+                self._send_email(self._build_email_body(prior_status, current_status))
+                prior_status = current_status
             else:
                 self.logger.trace('No changes in battery status.')
-            time.sleep(int(self.config['delay']))
 
-    def _get_composite_status(self):
-        """
-        Get status information about batteries connected to the device Batwatch is running on,
-         including the charge status, such that:
-        * If any battery is discharging, charge status is Discharging.
-          (Assume any status other than Charging or Fully Charged is equivalent to Discharging.)
-        * Charge status is Fully Charged only if all batteries are Fully Charged.
+            delay = random.uniform(0, self.config['average_delay'])
+            time.sleep(delay)
 
-        returns: CompositeStatus describing the count and overall charge status of the batteries.
+    def _get_composite_status(self):
+        """Get status information about batteries connected to the device Batwatch is running
+            on, including the charge status, such that:
+            * If any battery is discharging, charge status is Discharging.
+              (Assume any status other than Charging or Fully Charged is equivalent to
+               Discharging.)
+            * Charge status is Fully Charged only if all batteries are Fully Charged.
+
+        returns: CompositeStatus describing the count and overall charge status of the
+            batteries.
         """
 
         device_names = self.upower_bus.EnumerateDevices()
-        charge_status = FULLY_CHARGED
 
         batteries = []
         for device_name in device_names:
             device = self.system_bus.get(UPOWER_BUS_NAME, device_name)
-            if device.PowerSupply is True and device.Type in (2, 3):  # Type 2 = Battery, Type 3 = UPS
+            # We only care about device types 2 and 3, which are batteries and UPS units,
+            #   respectively. The full list can be found here:
+            #   https://upower.freedesktop.org/docs/Device.html#Device:Type
+
+            # A device is considered a power supply if it powers the whole system.
+            if device.PowerSupply is True and device.Type in (2, 3):
                 batteries.append(device)
 
+        charge_status = FULLY_CHARGED
+
         if not batteries:
             charge_status = NO_BATTERY
 
-        for battery in batteries:
-            if battery.State == 1:
-                charge_status = CHARGING
-            elif battery.State == 4 and charge_status != CHARGING:
-                charge_status = FULLY_CHARGED
-            else:
-                charge_status = DISCHARGING
-                break
+        else:
+            for battery in batteries:
+                if battery.State == 1:
```

Comment:
Are there any UPower constants we can refer to instead of hard coding these number? If not, should we put inline comments describing what the numbers bean? (I'm debating if that is too obvious.)

Reply:
I don't see a way to obtain the constant from UPower, as we are just doing D-Bus calls. Inline comments would probably work. We could also use an enum if we want more expressive code.

Reply:
Instead of enums, I guess you could just make a couple of all caps constants.